### PR TITLE
Refactor Storage

### PR DESCRIPTION
**Pull request by stechu**
This is part of refactoring `pallet-manta-pay` storage.

- [x] fix sharding function (use `Blake2s` to produce a random bit as the sharding function)
- [x] get rid of `manta-ledger`
- [x] improve the efficiencies of operations regards to a single shard

### CODE REVIEW

File: manta-data/Cargo.toml [hunk 1]
```diff
@@ -32,6 +32,6 @@ ark-std = { version = "0.3.0", default-features = false }
 # If you need to change any them, make sure to reflect the changes in whichever script is supposed to modify them.
 # Search for "Check Manta-Types" in Manta-Network repositories to find the relevant scripts.
 manta-asset = { path = "../manta-asset", default-features = false }
-manta-crypto = { branch = "manta", git = "https://github.com/Manta-Network/manta-crypto", default-features = false }
-manta-error = { branch = "manta", git = "https://github.com/Manta-Network/manta-error", default-features = false }
+manta-crypto = { path = "../../manta-crypto", default-features = false }
+manta-error = { path = "../../manta-error", default-features = false }
```
Comment (bhgomes) [inline manta-data/Cargo.toml:36]: These need to be changed back to the `git` dependencies.

File: manta-data/src/lib.rs [hunk 2]
```diff
@@ -26,10 +26,11 @@ mod test;
 
 use codec::{Decode, Encode};
 use manta_asset::{AssetBalance, AssetId, MantaAsset, MantaRandomValue, NUM_BYTE_ZKP};
-use manta_crypto::{AccountMembership, LedgerMerkleTreeRoot, MantaEciesCiphertext};
+use manta_crypto::{LedgerMerkleTreeRoot, LedgerPath, MantaEciesCiphertext};
 
 pub use metadata::BuildMetadata;
 
+/// Note: these values are still correct, however should not be used anymore since we are using parity codec to serdes
```
Comment (bhgomes) [inline manta-data/src/lib.rs:33]: If you want, you can mark this as deprecated:

```rust
#[deprecated = "These values are still correct, but should not be used anymore.
    We are using parity codec for serdes."]
/// ...
pub const MINT_PAYLOAD_SIZE: usize = 184;
```

**Review by bhgomes — CHANGES_REQUESTED** [review]
See `Cargo.toml` and other comment. LGTM.

**Review by Kevingislason — COMMENTED** [review]
LGTM. I suppose you know getting rid of `BuildMetadata` will break `manta-signer`, right?

**Comment by bhgomes** [comment]
> I suppose you know getting rid of `BuildMetadata` will break `manta-signer`, right?

Yeah, you're right. I forgot about this. I think it will also break `manta-api`, we need to keep it for now.

File: manta-data/src/lib.rs (hunk 2 above)
Comment (stechu) [reply manta-data/src/lib.rs:33]: How about just remove them?

Comment (bhgomes) [reply manta-data/src/lib.rs:33]: Yeah, I would just remove them.

**Comment by stechu** [comment]
The following function need to be implemented after the merkle tree implementation is merged into `manta-crypto`:
```rust
impl BuildMetadata for MantaAsset {
```
@bhgomes

**Comment by bhgomes** [comment]
See Manta-Network/manta-crypto/pull/27.

**Pull request merged**